src/main/scala/com/fulcrumgenomics/util/ReadStructure.scala

```diff
@@ -150,24 +141,64 @@ object ReadSegment {
     }
   }
 
-
   /** Creates a new read segment of the same type but with a new length */
   def apply(segment: ReadSegment, length: Int): ReadSegment = segment match {
     case seg: Template         => Template(seg.offset, length)
     case seg: SampleBarcode    => SampleBarcode(seg.offset, length)
     case seg: MolecularBarcode => MolecularBarcode(seg.offset, length)
     case seg: Skip             => Skip(seg.offset, length)
   }
+
+  def structureRead(segments: Seq[ReadSegment], bases: String, useOffset: Boolean = false): String = {
```

Comment:
Honestly I'm having a really hard time figuring out what it is this method is trying to do.  I think the name should probably be `extractBarcode` but even that's not clear to me.  A concise doc comment of what it is this method is doing would help me.